src/vectorchord/usage/measure-recall.md

```diff
@@ -22,12 +30,114 @@ SELECT vchordrq_evaluate_query_recall(query => $$
 $$, exact_search => true);
 ```
 
+## Query sampling <badge type="tip" text="since v0.5.2" />
+
+You may find it more useful to evaluate user-generated queries than randomly generated ones. VectorChord includes a built-in method to sample and store recent user queries. These records are grouped by index, making them easily accessible and seamlessly integrable with recall evaluation.
+
+```sql
+-- Enable the query sampling
+ALTER SYSTEM SET vchordrq.query_sampling_enable = on;
+
+-- Set the maximum number of queries to record per index
+ALTER SYSTEM SET vchordrq.query_sampling_max_records = 1000;
+
+-- Set the sampling rate (e.g., 1.0 for all queries, 0.01 for 1%)
+-- For high-traffic systems, a rate of 0.01 or lower is recommended.
+ALTER SYSTEM SET vchordrq.query_sampling_rate = 1;
+SELECT pg_reload_conf();
+
+-- Now, execute an example vector search
+-- SELECT * FROM items ORDER BY embedding <-> '[3,1,2]' LIMIT 10;
+```
+
+After queries have been executed and sampled, you can inspect the captured components of them, including the schema, index, table, column, operator, and vector.
+
+```sql
+-- List recorded queries for a specific index
+SELECT * from vchordrq_sampled_queries('idx');
+-- schema_name | index_name | table_name | column_name | operator |    value
+---------------+------------+------------+-------------+----------+--------------
+-- public      | idx        | items      | embedding   | <->      | [0.5,0.25,1]
+
+-- List recorded queries for all accessible indexes
+-- DANGER: The result may unexpectedly mix records from multiple indexes. Please check carefully before use. // [!code error]
+-- SELECT COUNT(DISTINCT index_name) FROM vchordrq_sampled_queries WHERE ... // [!code error]
+SELECT * from vchordrq_sampled_queries;
+```
+
+To measure the recall of the recorded queries, you can reconstruct the target query statement. The `vchordrq_sampled_queries` view is **not** recommended here because the evaluated recall will be incorrect if the records come from more than one index.
+
+```sql
+SELECT AVG(recall_value) FROM (
+    SELECT vchordrq_evaluate_query_recall(
+            format(
+                'SELECT ctid FROM %I.%I ORDER BY %I %s %s LIMIT 10',
+                quote_ident(lq.schema_name),
+                quote_ident(lq.table_name),
+                quote_ident(lq.column_name),
+                lq.operator,
+                quote_literal(lq.value)
+            )
```

Comment:
```suggestion
            format(
                'SELECT ctid FROM %I.%I ORDER BY %I %s %L LIMIT 10',
                lq.schema_name,
                lq.table_name,
                lq.column_name,
                lq.operator,
                lq.value
            )
```

https://www.postgresql.org/docs/current/functions-string.html

Reply:
```sql
                CASE
                    WHEN OP.opcname LIKE '%%l2%%' THEN '<->'
                    WHEN OP.opcname LIKE '%%ip%%' THEN '<#>'
                    WHEN OP.opcname LIKE '%%cosine%%' THEN '<=>'
                    ELSE ''
                END AS operator
```

I reviewed the code again for it. Please modify the code to avoid cheap judgments.